sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2150,9 +2150,6 @@ class Analyzer(override val catalogManager: CatalogManager)
                         lead.copy(ignoreNulls = ignoreNulls)
                       case lag: Lag =>
                         lag.copy(ignoreNulls = ignoreNulls)
-                      case _ =>
-                        throw QueryCompilationErrors.functionWithUnsupportedSyntaxError(
-                          owf.prettyName, "IGNORE NULLS")
```

Comment:
@beliefer I think you should keep this default match. I remember Scala 2.13 issues a warning about this.

Reply:
@beliefer  can you revert these removal here?

Reply:
`sealed` means we allow subclasses in the same source file. So, this is correct, @HyukjinKwon .

Reply:
Hm, but I thought we should keep the default match here. If `sealed` requires to remove the default match here, we shouldn't do that.

Reply:
For example, we could implement some new expression that implements `FrameLessOffsetWindowFunction` later (in the same file), and we forgot to fix it here that ends up with a raw case match error. I mean "match may not be exhaustive" warning .. like we fixed it https://github.com/apache/spark/pull/30455 before.

Or does it not show the warning in this case because we're using `sealed`?

Reply:
Scala compiler intelligently understands `sealed` semantics. That is a compilation failure in Scala 2.13.
> For example, we could implement some new expression that implements FrameLessOffsetWindowFunction later 

If you see the PR description of #30455 , I wrote like the following. Since we upgrade to Scala 2.13.4, we are safe unless the committer ignores GitHub Action job.
> Fix exhaustivity issues in both Scala 2.12/2.13 (Scala 2.13.4 requires this for compilation.)

Reply:
I think that the complier misses several cases to be honest (see https://github.com/apache/spark/pull/30923/files#diff-aad995a1ccfd7ca8e2a4c7bdf5114ddda8f9017e56cf8c60bb63846b3b5f3153R48-R50 as an example).
But I got that it's easier just to trust complier for maintenance.

Either way works to me since the complier doesn't complain for now.

Reply:
For the example, GitHub Action job is not triggered correctly due to 3rd party action (`linter` job's `setup-r` action). Since I recovered the GitHub Action, I remember it correctly.
![Screen Shot 2021-01-05 at 11 16 19 PM](https://user-images.githubusercontent.com/9700541/103740432-0ca45e80-4fac-11eb-9adf-683ae9ae2419.png)

There is no known cases where Scala 2.13 compiler miss. If there is a missing case, we should report to Scala community.